config/utils.go

```diff
@@ -35,22 +35,27 @@ func desanitize(actual, target *model.Config) {
 	if *target.SqlSettings.DataSource == model.FAKE_SETTING {
 		*target.SqlSettings.DataSource = *actual.SqlSettings.DataSource
 	}
-	if *target.SqlSettings.AtRestEncryptKey == model.FAKE_SETTING {
-		target.SqlSettings.AtRestEncryptKey = actual.SqlSettings.AtRestEncryptKey
+
+	if len(target.SqlSettings.DataSourceReplicas) == 1 && target.SqlSettings.DataSourceReplicas[0] == model.FAKE_SETTING {
```

Comment:
This is because before it was overwriting always this, so you can't change this value from the outside. Now it is de-sanitized only if was a FAKE_SETTING, it doesn't impact the interface (for now) because is not shown, but from the API perspective, suddenly you are receiving a `[*************************************]` in the Read replicas and Search replicas. I guess is not a big deal.